lblod/app-worship-decisions-database
# Fix/submissions ingest issues

## Description

This PR fixes an issue where worship-submissions wouldnt be visible for some provinces like Antwerp. The solution exists of 2 parts. Both a data issue and a dispatch issue in the worhsip-dispatcer service.

 ## Type of change


 ## Related services

 - [worship-submissions-graph-dispatcher-service](https://github.com/lblod/worship-submissions-graph-dispatcher-service)

 ## How to test

- clone and open both this app and worship-submissions-graph-dispatcher-service (worship-dispatcher)
- checkout the `bug/classification-prefix` branch for the worship-dispatcher service.
- add the followig docker-compose.overrride.yml
```
services: 
  submissions-dispatcher:
    image: lblod/worship-submissions-graph-dispatcher-service:0.8.0
    environment:
      NODE_ENV: "development"
    volumes:
      - "/path/to/worship-submissions-graph-dispatcher-service/:/app/"
  submissions-consumer:
    image: lblod/delta-consumer:0.0.15
    environment:
      DCR_SYNC_BASE_URL: "https://dev.loket.lblod.info/"
      DCR_SYNC_LOGIN_ENDPOINT: 'https://dev.loket.lblod.info/sync/worship-submissions/login' # don't forget DCR_SECRET_KEY in docker-compose.override.yml
      DCR_KEEP_DELTA_FILES: "true"
      DCR_SECRET_KEY: "ask-the-key"
      DCR_START_FROM_DELTA_TIMESTAMP: "2023-01-01T13:57:36.344Z"
      DCR_WAIT_FOR_INITIAL_SYNC: 'false'
      DCR_DISABLE_INITIAL_SYNC: "true"
```

be sure to replace the volume of submissions-dispatcher to your path and ask the secret key for consumer in chat.

- start app-worship-decision-database and run `docker-compose logs --tail=100 -f submissions-consumer`
- login as a eredienst bestuur (e.g. Kerkfabriek O.-L.-Vrouw van Deinze ) in dev loket and create toezicht submissions (e.g notulen)
- wait for stuff to happen in submissions-consumer logs
- check if provinces can see submissions of their linked eredienst besturen
 (Kerkfabriek O.-L.-Vrouw van Deinze is linked to provincie antwerpen)
 
 note: dispatch rules have test examples you can use 
 https://github.com/lblod/worship-submissions-graph-dispatcher-service/blob/master/dispatch-rules/notulen.js
